### ISSUE TITLE
LSP should recognise variables that are set using the "set suite variable" and "set global variable" keywords

### ISSUE THREAD

**Issue by Serhiy1** [robocorp/robotframework-lsp#577]
**Is your feature request related to a problem? Please describe.**
LSP allows you to go to the definition of variables that are created locally, set the `*** variables***` section and in some cases, variables defined in other files.

LSP should also recognise variables set with the  `set suite variable` and `set global variable`

**Describe the solution you'd like**
LSP should make an attempt at trying to find the definitions of these variables, barring from impossibilities, such as dynamically defined variables. 

**Describe alternatives you've considered**
A workaround that I currently use is to add a copy of the variables in the `***variables***` section, defining them to `${EMPTY}` and then adding a comment to direct the user where to look.  

Though this is not possible for `set global variable`

**Comment by fabioz** [comment]
Agreed. Can you provide a sample code on how you use those? (just to make sure I cover the actual use-case you're using)

**Comment by Serhiy1** [comment]
Sure here is a quick example

file 1
```robot
*** Settings ***
Documentation    Example of using suite variables
Resource         file_2.resource
Suite Setup         Create Suite Resources

*** Test Cases ***
Test 1
    [Documentation]    Example of suite variable

    Log To Console    ${CONST_1}

Tests Global
    [Documentation]    Example of global variable
    Log To Console    ${GLOBAL_VAR}
    

*** Keywords ***
Create Suite Resources
    [Documentation]    Sets up some suite variables

    Set Suite Variable    ${CONST_1}    foo
```

File 2

```robot

*** Settings ***
Documentation    Example of using suite variables
Suite Setup         Create Suite Resources

*** Keywords ***
Create Suite Resources
    [Documentation]    Sets up some suite variables

    Set Global Variable    ${GLOBAL_VAR}    bar

```

**Comment by fabioz** [comment]
As a note, this is now available in a pre-release (0.42.2).

See: https://github.com/robocorp/robotframework-lsp/blob/master/robotframework-ls/docs/changelog.md for all that's changed in the pre-release.
